src/Shared/SqlProcessor.cs

```diff
@@ -161,58 +585,86 @@ private static bool SanitizeHexLiteral(string sql, ref int index)
                 {
                     continue;
                 }
+#endif
 
                 i -= 1;
                 break;
             }
 
-            index = i;
+            state.ParsePosition = ++i;
+
+            buffer[state.SanitizedPosition++] = '?';
             return true;
         }
 
         return false;
     }
 
-    private static bool SanitizeNumericLiteral(string sql, ref int index)
+    private static bool SanitizeNumericLiteral(ReadOnlySpan<char> sql, Span<char> buffer, ref ParseState state)
     {
-        var i = index;
-        var ch = sql[i];
+        var i = state.ParsePosition;
+        var nextChar = sql[i];
```

Comment:
`nextChar` kind of reads to me like not the one we're looking at now, but the one to come after. Then there's access to `i + 1` which feels to me like next, but we're already calling `i` next, so that would be `next next`.

Maybe this could be renamed to `thisChar`/`currentChar`?